### ISSUE TITLE
Bugs with Autocompletion and calltips

### ISSUE THREAD

**Issue by anseok** [leo-editor/leo-editor#3743]
PR #3805 fixes this issue.

**Original report**

I am following the [tutorial](https://leo-editor.github.io/leo-editor/tutorial-basics.html) using Leo 6.7.6, Python 3.11.2, PyQt version 5.15.8, linux.  In the section "Autocompletion and calltips", I've encountered two bugs.

On autocompletion, the tutorial says "typing ‘?’ will show the docstring for a method."  However, doing so logs a stack trace that ends with

```
  File "/opt/local/lib/python3.11/dist-packages/leo/core/leoKeys.py", line 827, in info
    argspec = inspect.getargspec(obj)
              ^^^^^^^^^^^^^^^^^^
AttributeError: module 'inspect' has no attribute 'getargspec'
```

This appears to be due to Python 3.11 having dropped inspect.getargspec method.  Perhaps using instead [inspect.getfullargspec](https://docs.python.org/3.11/library/inspect.html#inspect.getfullargspec) would work for this version of Python.

Another bug is with calltips.  Following the instruction, I pressed Alt-2 (toggle-calltips) to activate calltips and tried the three examples in the documentation.  I could see no calltips with any of them, and there was no indication of errors in the log.

In case it helps to reproduce the behaviour, I am running Leo in a Docker container defined by the following Dockerfile

```
FROM debian:12-slim

# The dependencies are inferred from
# https://leo-editor.github.io/leo-editor/installing.html#id1

# python3-pyqt5.qtwebengine pulls in python3-pyqt5 and
# python3-sphinx pulls in python3-docutils.

RUN apt-get update;\
    apt-get -y --no-install-recommends install\
      fonts-noto-cjk-extra gcc ipython3 libc6-dev libenchant-2-2\
      pip python3-pyqt5.qtwebengine python3-sphinx python3-tk;\
    pip install --prefix=/opt leo;\
    useradd -mg users leo

USER leo
WORKDIR /home/leo/.leo
ENV PATH="$PATH:/opt/local/bin"\
    PYTHONPATH=/opt/local/lib/python3.11/dist-packages
ENTRYPOINT ["leo", "--ipython"]
```

**Comment by edreamleo** [comment]
@anseok Thanks for your report.

For Python 3.12, the inspect module raises `TypeError` when inspecting bound methods!  The crash is:
```
TypeError: 'method' object is not iterable
```
This looks like a Python bug to me. It's hard to believe this behavior is intended.  I'll investigate.

**Comment by edreamleo** [comment]
@anseok I have filed [Python bug #115630](https://github.com/python/cpython/issues/115630).

There seems to be nothing more I can do until this issue is resolved.

**Comment by edreamleo** [comment]
@anseok The workaround: catch the TypeError and fail gracefully :-)  I'll do that first.

**Comment by edreamleo** [comment]
@anseok The problem with calltips is a direct result of the problems with the inspect module.

Nothing more needs to be done because Leo already fails gracefully for calltips.

**Comment by edreamleo** [comment]
@anseok I'll create a new PR that uses `inspect.signature` and `inspect.docstring`.

**Comment by edreamleo** [comment]
Fixed via PR #3805.
